src/components/ui/button.tsx

```diff
@@ -37,17 +38,37 @@ export interface ButtonProps
   extends React.ButtonHTMLAttributes<HTMLButtonElement>,
     VariantProps<typeof buttonVariants> {
   asChild?: boolean;
+  loading?: boolean;
+  loadingText?: string;
 }
 
 const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
-  ({ className, variant, size, asChild = false, ...props }, ref) => {
-    const Comp = asChild ? Slot : "button";
+  (
+    {
+      className,
+      variant,
+      size,
+      children,
+      disabled,
+      loading,
+      loadingText,
+      ...props
+    },
+    ref,
+  ) => {
+    const isLoading = loading === true;
+    const isDisabled = disabled ?? isLoading;
+
     return (
-      <Comp
```

Comment:
Replace `radix-slot` with `button element` to avoid passing attributes like `disabled` to the child while using props like `loading` with different `asChild` child.